tests/pipelines/main.nf.test

```diff
@@ -37,6 +35,22 @@ nextflow_pipeline {
             def expected_metadata = path("$baseDir/tests/data/metadata/expected_merged_data.tsv")
             assert actual_metadata.text == expected_metadata.text
 
+            // Check auto-built config file:
+            def actual_config = path("$launchDir/results/build/config.json")
+            def expected_config = path("$baseDir/tests/data/configs/autoconfig_samplesheet.json")
+            assert actual_config.text == expected_config.text
+
+            // Arborator outputs:
+            // NOTE: Arborator produces a (somewhat) non-deterministic cluster summary,
```

Comment:
In what way is it non-deterministic? Is it related to the order samples are processed in channels? Since we could enforce a certain order in order to make deterministic in that case.

Reply:
So, what happens is that I can specify in the arborator config file the order of some columns, but then arborator will add a bunch of summary columns (automatically and unspecified) that are derived from the "main" columns.

For example, if I build an arborator config file that has the following `grouped_metadata_columns`: `outbreak`, `organism`, `subtype`, `country`, `serovar`, `age`, `date`, `source`, `special`, then I'll get an `arborator/cluster_summary.tsv` file that has these headers:

```
Outbreak Code   Organism        Subtype Country of Collection   Serovar Patient Age (years)     Date    Source Type     Special age_mean_value  count_country_The United States   count_serovar_O125      count_subtype_EAEC      date_min_value  count_source_cheese     count_source_fruit      max_dist        count_age_55    count_age_61      count_members   count_serovar_O111:H21  count_subtype_EPEC      count_age_21    count_age_35    age_min_value   min_dist        count_outliers  count_serovar_O126:H27    count_country_France    count_special_False     count_organism_Escherichia coli count_special_True      count_age_14    count_source_beef         count_country_Canada    date_max_value  count_serovar_O157:H7   count_subtype_EHEC/STEC count_age_43    count_source_milk       median_dist     age_max_value     age_median_value        mean_dist
```

However, if I run the workflow again, I'll get:

```
Outbreak Code   Organism        Subtype Country of Collection   Serovar Patient Age (years)     Date    Source Type     Special count_special_True      count_country_The United States   age_min_value   mean_dist       median_dist     count_subtype_EAEC      count_source_beef       count_source_milk       age_max_value   count_organism_Escherichia coli   count_serovar_O125      count_country_France    count_serovar_O157:H7   count_country_Canada    max_dist        count_age_55    count_source_cheese       count_source_fruit      count_subtype_EHEC/STEC age_mean_value  age_median_value        count_members   count_age_14    date_min_value  count_serovar_O111:H21    count_special_False     count_age_21    count_age_43    count_age_35    count_serovar_O126:H27  count_outliers  count_subtype_EPEC      min_dist  date_max_value  count_age_61
```

Basically, everything after my specifications (which are mapped from keys to labels: `outbreak` -> `Outbreak Code`), which means `Special` onwards, are effectively non-deterministic. I can't guarantee the ordering of these columns, so I'm only testing the part that is deterministic.

Reply:
Okay. Thanks so much for the explanation. There doesn't seem to be an easy way to make the output deterministic so what you are doing makes sense.